lib/3scale/backend/storage_helpers.rb

```diff
@@ -50,7 +50,8 @@ class << self
               # there's a timeout: https://github.com/redis/redis-rb/issues/668
               # We should investigate if there are edge cases that can lead to
               # duplicated commands because of this setting.
-              reconnect_attempts: 1,
+              reconnect_attempts: ThreeScale::Backend.listener? ? 1 : Float::INFINITY,
```

Comment:
I remember you saying something along the lines of 
> sync client takes this as an array of seconds to wait between reconnection attempts. Will this `INFINITY` be number of seconds sync client will wait?

Reply:
No, it will be the amount of retries, which no wait time, so infinite loop until redis comes back. However, that's the theory, in practice, I observed how the moment the connection drops the worker immediately crashes in sync mode. I ignored this because that's how it behaved before and nobody complained, I guess because OpenShift enters on a crash loop until the connection is restored, so at the end it will end up recovering the service when redis is back.

However I wonder why it crashes when it should just enter an infinite loop. I'll investigate...

Reply:
Yes, this was a bug. `Float` was not being accepted as a valid value. Fixed on https://github.com/3scale/apisonator/pull/412/commits/6a305de85c6504e5b052503623ad40280dd7a8e2